### DISCUSSION TITLE
Is there an extension mechanism for UI elements?

### DISCUSSION THREAD

**Discussion by SamyAteia** [asreview/asreview#1603] [Q&A]
Hi, is it possible to write extensions for the UI of asreview?
Example:
- Highlighting features in the abstract of papers
- Adding additional input to the initialization phase of a model, such as a review protocol, that can be used to extract better features.

Is there a build in extension mechanism that handles core updates gracefully, or do I have to fork and manually merge ASreview if I want to implement these features?

**Comment by Rensvandeschoot** [comment]
Great you are thinking of extending the ASReview software! 

Highlighting features in the abstract of papers is a long-wanted feature (see also https://github.com/asreview/asreview/discussions/995). So, if you would like to develop this feature, we would be open to adding it to the core! You might want to start a PR, and we are happy to provide feedback! 

For your second suggestion, you might want to wait for this PR to be merged: https://github.com/asreview/asreview/pull/1224 (@J535D165 any idea when it is expected??). This new process of setting up a project makes it really easy to add a custom text field to be used for training data. For example, to train the first iteration of the model with inclusion and exclusion criteria and many other variations. If you want to contribute by adding these features, this would be very much appreciated!!

**Reply by SamyAteia** [reply]
Thanks for the pointers! So I guess there are no direct extension mechanisms for UI elements, but there is general interest in including the suggested features in the main project. I will try to extend the core system and contribute the changes back.

**Reply by J535D165** [reply]
There is, indeed, no framework to extend the UI (would be nice to have). And yes, we are welcoming PRs that add features like this. For the second feature request, please branch of PR #1224 if not merged yet. 

Let us know if you need some help!
